# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/nsCSSProps.h
# Commit: 63603267402e
# Full Hash: 63603267402ee93d6551ef60c35191d99aaee06b
# Author: Zach Hoffman <zach@zrhoffman.net>
# Date: 2023-12-18 17:17:57
# Regressor Bug: 1846516
# File Overlap Count: 3
# Description:
#   Bug 1846516 - [css-properties-values-api] Introduce AnimatedPropertyID/AnimatedPropertIDSet. r=firefox-animation-reviewers,firefox-style-system-reviewers,layout-reviewers,emilio,hiro
#   
#   This will make possible to animate custom properties on Gecko side. For now, the
#   animation code keeps only dealing with nsCSSPropertyID, so behavior is unchanged.
#   
# ==============================================================================

diff -r 753bc39f59ff -r 63603267402e layout/style/nsCSSProps.h
--- a/layout/style/nsCSSProps.h	Mon Dec 18 09:24:07 2023 +0000
+++ b/layout/style/nsCSSProps.h	Mon Dec 18 09:24:08 2023 +0000
@@ -67,9 +67,12 @@
   static bool IsCustomPropertyName(const nsACString& aProperty);
 
   static bool IsShorthand(nsCSSPropertyID aProperty) {
+    if (aProperty == eCSSPropertyExtra_variable) {
+      return false;
+    }
     MOZ_ASSERT(0 <= aProperty && aProperty < eCSSProperty_COUNT,
                "out of range");
-    return (aProperty >= eCSSProperty_COUNT_no_shorthands);
+    return aProperty >= eCSSProperty_COUNT_no_shorthands;
   }
 
   // Same but for @font-face descriptors